codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoSettings.java

```diff
@@ -45,6 +47,13 @@ public final class GoSettings {
     private Boolean generateGoMod = false;
     private String goDirective = GoModuleInfo.DEFAULT_GO_DIRECTIVE;
     private ShapeId protocol;
+    private ArtifactType artifactType;
+
+    @SmithyInternalApi
+    public enum ArtifactType {
+        CLIENT,
+        SERVER;
+    }
```

Comment:
I don't think `ArtifactType` is doing anything meaningful right now, it can be removed.